### ISSUE TITLE
STICKERS DoS!!!! (SERIOUS)

### ISSUE THREAD

**Issue by nastys** [DrKLO/Telegram#1069]
Since version 2.8, users can now send stickers via the new, separate stickers panel.
This also means that anyone can send thousands of stickers in a row, thus sending thousands of notifications to someone's phone!!!!
This does not affect who upgraded to version 2.8 since smart notifications have been introduced as well, but IT STILL AFFECTS WHO HAVEN'T UPDATED YET!!!
So, someone who has Telegram 2.8 can send A LOT OF NOTIFICATIONS TO PEOPLE THAT HAVE TELEGRAM <=2.7!!!!

https://www.youtube.com/watch?v=bY5fMzptuIg

Fixing is EASY. Just close the keyboard after sending a sticker. (or add a delay between stickers)

@DrKLO PLEASE FIX THIS AND RELEASE VERSION 2.8.2!!
Since we have the separate panel, it happens so often that someone send a lot of stickers, and it's so annoying, even if I upgraded to version 2.8, which has smart notifications! :'(

EDIT: just to clarify, this does not affect text chats, because you don't have the time to write and send a message repeatedly within a second. Also, I need these people, so I can't just block them.

**Comment by AUDI-A4** [comment]
what about normal chat

**Comment by nastys** [comment]
@AUDI-A4 ??

**Comment by tsubus** [comment]
just block the person that is spamming. spamming has always been possible with text or regular emojis, this should not be fixed client-side, it's a user issue.

**Comment by nastys** [comment]
@tsubus I don't think you got it. These people are part of a group and they think it's funny to send a lot of notifications to other people in the same group. I don't really want to kick them, just stop them from sending thousands of stickers in a row. As I stated before, you can't just write a message then send it hundreds of times in a second, so the problem exists only with stickers. It's true that you could also send a lot of stickers in a row with the web client, however they don't even know it, so I don't care that much.

**Comment by tsubus** [comment]
I don't think user stupidity should be blocked by the client, just invite smarter people to your group.

**Comment by nastys** [comment]
@tsubus They're my classmates and we need Telegram to exchange files. That's why I can't kick them.

**Comment by auchri** [comment]
Then tell them that they should stop or they will be kicked form the group.

**Comment by nastys** [comment]
@auchri They know I won't kick them after I worked so hard to make them install Telegram. The best I can do is kick them temporarily and persuade all others to update Telegram... but I'd prefer to avoid that.

**Comment by frazzmark** [comment]
Well you could just add that after entering a sticker, the keyboard closes, which kind of makes dense also.

**Comment by nastys** [comment]
@frazzmark That's a nice idea, added, thanks! :D

**Comment by jomo** [comment]
Spam != DoS

**Comment by nastys** [comment]
@jomo Well, if you can't even use your phone because of the notifications...

**Comment by unaizalakain** [comment]
It would be nice to have an option to turn off stickers and display their related emojis (the ones that are displayed in the chat summary) instead though, many group chat users would greatly appreciate it.

**Comment by Walkman100** [comment]
> It would be nice to have an option to turn off stickers and display their related emojis

Now THAT'S an idea.

**Comment by joshery420** [comment]
@unaizalakain Unfortunately that wouldn't solve @nastys issue. Spamming messages is the same whether it's a sticker or an emoji. 
Assuming this is still an issue for you, why not just mute the group conversation when you're not using it? It's not really the job of the app to stop people from being annoying, and if you can't block them, just mute the whole conversation until you actually need it.
Also, pretty sure the "smart notifications" feature doesn't do what you're thinking, but nonetheless, why would you continue running an older version?

**Comment by nastys** [comment]
@joshery420 Spamming stickers is much quicker than spamming emojis. Also, of course I upgraded. It's not for me, it's for my friends who don't care about upgrades.

**Comment by joshery420** [comment]
@unaizalakain I suppose because with an emoji you still have to hit send, so yes you're right, it's technically less spammable than stickers.
Also, wasn't harassing you about not updating, was just genuinely curious haha. I misunderstood about who wasn't updated, so I assumed it was you since you were complaining about not having the smart notifications. My mistake! ;)
All that aside, I still stand by my previous suggestion of just muting those specific group chat(s) when you're not using them. It totally sucks that your friends are being obnoxious, but I also don't think it's the app or developer's job to prevent that.

**Comment by nastys** [comment]
@joshery420 What happened to my group is just an example. Even if everyone updated and/or muted the chat, if someone sent a lot of stickers, the chat would be flooded with a lot of messages. To be honest I don't see why not fix it, as it doesn't seem that hard to add a 1-sec. delay between each sticker/message.

**Comment by joshery420** [comment]
@nastys I'm pretty confident that some people spamming stickers is annoying you and your friends much more than it's causing issues for Telegram's servers. https://telegram.org/blog/2-billion

Suffice it to say that I highly doubt either a 1-sec delay between sending messages/stickers or closing the stickers tray after sending each sticker would really do anything to deter someone who wants to send a flood of stickers anyway. Sure it'd slow them down a little bit, but not much, and I'd wager a number of people (myself included) would find the change more annoying than anything.

It's not a matter of the "feature" being hard to implement, it's just that I strongly believe that a good messaging client should _never arbitrarily stop people from messaging_, especially just because someone finds it irritating. Telegram is a tool for people to use, not a gatekeeper telling people how they're allowed to communicate. Too many messages between times you look at the thread? That's why Telegram automatically scrolls to the last message you read so you know where to start reading to catch up.

Like I said before, if your concern is over the constant notifications, just mute the chat when you're not actively using it, or maybe just ask your friends/co-workers/classmates to stop spamming stickers in the thread. Hopefully they have a bit of respect and they'll back off. Maybe you could even show them how to make another group chat with just the people who don't mind stickers being spammed, letting them be annoying without bothering anyone else haha. :wink: 

Also, sorry if it sounds like I'm being mean or anything. Absolutely not trying to be rude, I just have strong feelings on the topic. What you're asking for kinda reminds me of chat threads in some games that prevent you from messaging more than once every few seconds, which can make holding a real-time conversation irritatingly slow. If people want to message fast, be it with text, emoji, or stickers, I say let them.

**Comment by DRSDavidSoft** [comment]
Is there a way to block a certain sticker? Some of them are quite nasty!
Also, it would be awesome to block all stickers from a certain people.

I really love the freedom of Telegram stickers, but I also hate when some idiots abuse it by sending really nasty or disgusting stickers. I don't know why they think that's funny.

Also, is there a way to block someone's message in a group? As far as I know, blocking him will prevent him sending messages to *me*, not to groups. I know I can leave the group, but I don't want to! (: I just want his messages and/or stickers to be hidden while I read the group's messages.
